doc/source/security/selinux.rst

```diff
@@ -0,0 +1,507 @@
+
+    SELinux is preventing mysqld from add_name access on the directory mysqld.log.
+
+SELinux prevents mysql from creating a log file inside the directory. 
+
+You can view all the violations by changing the SELinux mode to ``Permissive`` and then running mysqld. All violations are logged in the journal log. After this run, you can generate a local policy module, install it, and switch SELinux back to ``enforcing`` mode.
+
+1. Unload the current local my-mysqld policy module:
+
+    .. code-block:: bash
+
+        $ semodule -r my-mysqld
+
+2. You can put a single domain into permissive mode. This option allows the other domains on the system to remain in enforcing mode. Use ``semanage permissive`` with the ``-a`` parameter to change mysqld_t to permissive mode:
+
+    .. code-block:: bash
+
+        $ semanage permissive -a mysqld_t
+
+3. Verify the mode change:
+
+    .. code-block:: bash
+
+        semdule -l | grep permissive
+        ...
+        permissive_mysqld_t
+        ...
+
+4. To make searching the log easier, return the time:
+
+    .. code-block:: bash
+
+        $ date
+
+5. Start the service.
+
+    .. code-block:: bash
+
+        $ service mysqld start
+
+6. MySQL starts and SELinux logs all of the violations in the journal log. Check the journal log:
+
+    .. code-block:: bash
+
+        $ journalctl -xe
+
+7. Stop the service:
+
+    .. code-block:: bash
+
+        $ service mysqld stop 
+
+8. Generate a local mysqld policy:
+
+    .. code-block:: bash
+
+        $ ausearch --start 10:00:00 -c 'mysqld' --raw | audit2allow -M my-mysqld
+
+9. Review the policy (the policy you generate may be different):
+
+    .. code-block:: bash
+
+        $ cat my-mysqld.te
+        module my-mysqld 1.0;
+
+        require {
+        type default_t;
+            type mysqld_t;
+            class dir { add_name write };
+            class file { append create open };
+        }
+
+        #============= mysqld_t ==============
+        allow mysqld_t default_t:dir { add_name write };
+        allow mysqld_t default_t:file { append create open };
+
+10. Install the policy:
+
+    .. code-block:: bash
+
+        $ semodule -i my-mysqld.pp
+
+11. Use ``semanage permissive`` with the ``-d`` parameter, which deletes the permissive domain, to returns the domain to enforcing:
+
+    .. code-block:: bash
+
+        $ semanage permissive -d mysqld_t
+
+12. Restart the service:
+
+    .. code-block:: bash
+
+            $ service mysqld start
+
+.. note::
+
+    You can use this procedure to adjust the local mysqld policy module. You should review the changes generated to ensure the rules are not too permissive.
+
+.. _selinux-secure-file-priv:
+
+Setting ``secure_file_priv`` directory
+---------------------------------------
+
+Depending on the |Percona Server| version, you may be required to set the SELinux tags on the ``/var/lib/mysql-files/`` directory. This directory is used for ``SELECT ... INTO OUTFILE`` or similar operations. 
```

Comment:
I would add the info that mysqld simply needs r/w access to the destination directory and this has to be allowed by SELinux rules.

BTW, where from we have got this info about "depending on the ..."?

Reply:
Rewritten